src/contract/assignments.rs

```diff
@@ -150,6 +151,25 @@ impl ToMerkleSource for OwnedRightsInner {
     }
 }
 
+impl MergeRevealed for OwnedRights {
+    fn merge_revealed(&mut self, other: &Self) -> Result<bool, String> {
+        if OwnedRightsInner(self.clone())
```

Comment:
We should not use clone while we did the func `&mut self` to avoid cloning=memory allocations! `to_merke_source` does not need to own the data, so probably it will be better to have `MergeRevealed` implemented for inner type...

Reply:
Yes understood. Changing the trait signature to take ownership of the data to avoid any cloning. I hope taking ownership of both the data will not cause many problems downstream.